### ISSUE TITLE
Markdown footnotes surrounding a word result in superscript

### ISSUE THREAD

**Issue by adunning** [jgm/pandoc#8652]
**Explain the problem.**
Surrounding a word with footnotes in Markdown results in a superscript instead of the notes.

```shell
pandoc << EOT
^[note1]test^[note2]
EOT
```

Result:

```html
<p><sup>[note1]test</sup>[note2]</p>
```

**Pandoc version?**
pandoc 3.1, macOS 13.2.1

**Comment by jgm** [comment]
`^[note1]test^` happens to be pandoc markdown syntax for superscript. So there's a syntactic ambiguity here and pandoc is preferring the first match.  This might be a case where you need to use a regular footnote, not the inline form.

**Comment by adunning** [comment]
This also occurs with two inline footnotes in a row:

```shell
pandoc << EOT
test^[note1]^[note2]
EOT
```

As you note, regular footnotes solve both cases:

```shell
pandoc << EOT
[^1]test[^2]

test[^1][^2]

[^1]: note1
[^2]: note2
EOT
```

Is there any case to be made for having the footnotes take precedence? The superscripts could still work with the square brackets escaped, `test^\[note1\]^\[note2\]`.

**Comment by jgm** [comment]
Yes, it probably makes sense to make footnotes take precedence here.

**Comment by jgm** [comment]
The code has the following comment:
```
     '^'     -> superscript <|> inlineNote -- in this order bc ^[link](/foo)^
```
So apparently it was an intentional decision to give superscripts priority.

**Comment by jgm** [comment]
An alternative would be to say that an inline footnote can't be followed directly by a `(` or `[`.

**Comment by chaojian-zhang** [comment]
There is some bug (rather than mere priority) component of this issue, as commented here: https://github.com/jgm/pandoc/issues/5878#issuecomment-3294673480

When footnote contains words (with spaces), same problem happens.

Regular footnote is a pain to maintain when updating (e.g. inserting) footnotes, which requires updating all indices (or making up artificial names).

**Comment by rnwst** [comment]
See also #11054, which is a duplicate of this issue.
